source/core/ut_suite_manager.pkb

```diff
@@ -118,46 +122,47 @@ create or replace package body ut_suite_manager is
             if l_proc_annotations.exists('testteardown') then
               l_teardown_procedure := ut_annotations.get_annotation_param(l_proc_annotations('testteardown'), 1);
             end if;
-            
+          
             if l_proc_annotations.exists('rollback') then
               l_rollback_annotation := ut_annotations.get_annotation_param(l_proc_annotations('rollback'), 1);
-              l_rollback_type := case lower(l_rollback_annotation) 
-                                   when 'manual' then
-                                     ut_utils.gc_rollback_manual
-                                   when 'auto' then
-                                     ut_utils.gc_rollback_auto
-                                   --when 'on-error' then
-                                   --  ut_utils.gc_rollback_on_error
-                                   else
-                                     l_suite_rollback
-                                   end;
+              l_rollback_type       := case lower(l_rollback_annotation)
+                                         when 'manual' then
+                                          ut_utils.gc_rollback_manual
+                                         when 'auto' then
+                                          ut_utils.gc_rollback_auto
+                                       --when 'on-error' then
```

Comment:
not now. this remindes about the third type of transaction control. it might became possible to implement after your changes to `ut_executable`